src/Data/Bin/Ordering.agda

```diff
@@ -0,0 +1,791 @@
+1+[2x]<2[1+x] x =  odd<even (inj₂ refl)
+
+<-irrefl : Irreflexive _≡_ _<_
+
+<-irrefl {zero}     {zero}      _  ()
+<-irrefl {zero}     {2[1+ _ ]}  ()
+<-irrefl {zero}     {1+[2 _ ]}  ()
+<-irrefl {2[1+ _ ]} {zero}      ()
+<-irrefl {2[1+ x ]} {2[1+ .x ]} refl =  x'≮x'
+  where
+  x'≮x' : 2[1+ x ] ≮ 2[1+ x ]
+  x'≮x' (even<even x<x) =  <-irrefl refl x<x
+
+<-irrefl {1+[2 x ]} {1+[2 .x ]} refl =  x'≮x'
+  where
+  x'≮x' : 1+[2 x ] ≮ 1+[2 x ]
+  x'≮x' (odd<odd x<x) =  <-irrefl refl x<x
+
+<⇒≢ :  _<_ ⇒ _≢_
+<⇒≢ {x} {_} x<y x≡y =  <-irrefl {x} {x} refl x<x
+  where
+  x<x = subst (x <_) (sym x≡y) x<y
+
+>⇒≢ :  _>_ ⇒ _≢_
+>⇒≢ {_} {_} y<x =  ≢-sym (<⇒≢ y<x)
+
+≡⇒≮ :  _≡_ ⇒ _≮_
+≡⇒≮ x≡y x<y =  <⇒≢ x<y x≡y
+
+≡⇒≯ :  _≡_ ⇒ _≯_
+≡⇒≯ x≡y x>y =  >⇒≢ x>y x≡y
+
+<-trans : Transitive _<_
+<-trans {_}      {_}      {zero}     _  ()
+<-trans {zero}   {_}      {2[1+ _ ]} _  _  =  0<even
+<-trans {_}      {zero}   {_}        ()
+<-trans {zero}   {_}      {1+[2 _ ]} _  _  =  0<odd
+
+<-trans {2[1+ x ]} {2[1+ y ]} {2[1+ z ]} (even<even x<y) (even<even y<z) =
+                                                    even<even (<-trans x<y y<z)
+<-trans {2[1+ x ]} {2[1+ y ]} {1+[2 z ]} (even<even x<y) (even<odd y<z) =
+                                                    even<odd (<-trans x<y y<z)
+<-trans {2[1+ x ]} {1+[2 y ]} {2[1+ z ]} (even<odd x<y) (odd<even (inj₁ y<z)) =
+                                                    even<even (<-trans x<y y<z)
+<-trans {2[1+ x ]} {1+[2 y ]} {2[1+ .y ]} (even<odd x<y)
+                                          (odd<even (inj₂ refl)) =  even<even x<y
+<-trans {2[1+ x ]} {1+[2 y ]} {1+[2 z ]} (even<odd x<y) (odd<odd y<z) =
+                                                        even<odd (<-trans x<y y<z)
+<-trans {1+[2 x ]} {2[1+ y ]} {2[1+ z ]} (odd<even (inj₁ x<y)) (even<even y<z) =
+                                                   odd<even (inj₁ (<-trans x<y y<z))
+<-trans {1+[2 x ]} {2[1+ .x ]} {2[1+ z ]} (odd<even (inj₂ refl))
+                                          (even<even x<z) =  odd<even (inj₁ x<z)
+<-trans {1+[2 x ]} {2[1+ y ]} {1+[2 z ]} (odd<even (inj₁ x<y)) (even<odd y<z) =
+                                                         odd<odd (<-trans x<y y<z)
+<-trans {1+[2 x ]} {2[1+ .x ]} {1+[2 z ]} (odd<even (inj₂ refl)) (even<odd x<z) =
+                                                                 odd<odd x<z
+<-trans {1+[2 x ]} {1+[2 y ]} {2[1+ z ]} (odd<odd x<y) (odd<even (inj₁ y<z)) =
+                                                  odd<even (inj₁ (<-trans x<y y<z))
+<-trans {1+[2 x ]} {1+[2 y ]} {2[1+ .y ]} (odd<odd x<y) (odd<even (inj₂ refl)) =
+                                                             odd<even (inj₁ x<y)
+<-trans {1+[2 x ]} {1+[2 y ]} {1+[2 z ]} (odd<odd x<y) (odd<odd y<z) =
+                                                       odd<odd (<-trans x<y y<z)
+
+<⇒≯ : _<_ ⇒ _≯_
+<⇒≯ {x} {y} x<y y<x =  <-irrefl refl (<-trans {x} {y} {x} x<y y<x)
```

Comment:
The fact that `x` and `y` aren't inferable here suggests that the definition of `<` isn't quite right.

Reply:
In fact, they are inferrable
(if this means that it is possible to omit implicit arguments here). 
I have changed the implementations to 
```
<⇒≯ x<y y<x =  <-irrefl refl (<-trans x<y y<x)
...
>⇒≮ y<x =  <⇒≯ y<x
```
and they are type-checked all right.